### ISSUE TITLE
redux-form should not write errors to the console explicitly.

### ISSUE THREAD

**Issue by naw** [erikras/redux-form#3821]
Formerly https://github.com/erikras/redux-form/pull/3723 was merged to address https://github.com/erikras/redux-form/issues/3636

The original issue (https://github.com/erikras/redux-form/issues/3636) was that exceptions thrown in an `onSubmitSucceeded` handler were being swallowed. The fix in https://github.com/erikras/redux-form/pull/3723 handled this by explicitly logging an error to the console (i.e. with `console.error`)

I'd like to suggest that this isn't the correct fix for the problem.

When an error is thrown within a promise, the normal and expected behavior is for the promise to reject. This was occurring, but due to a separate issue (https://github.com/erikras/redux-form/issues/3303), the promise was later being marked as resolved.

The _correct_ fix to the original problem is to let the error in an `onSubmitSucceeded` handler bubble up as an unhandled rejection. This has already been solved in https://github.com/erikras/redux-form/pull/3744, although it wasn't solved at the time that https://github.com/erikras/redux-form/issues/3636 was opened.

In other words, the fix in https://github.com/erikras/redux-form/pull/3744 makes the fix in https://github.com/erikras/redux-form/pull/3723 unnecessary, and also the fix (explicit use of `console.error`) in https://github.com/erikras/redux-form/pull/3744 is undesirable. We should not consider exceptions raised in a promise to be errors that should be written to the console -- instead, just let them bubble up as rejected promises.

I'm opening this issue to request that you consider reverting https://github.com/erikras/redux-form/pull/3723

Thanks!

**Comment by naw** [comment]
cc/ @h1b9b  @ClementParis016

**Comment by naw** [comment]
Note: The current code will `console.error` not just for errors thrown in `onSubmitSucceeded`, but for errors thrown in `onSubmit` also. So supposing you return a promise from `onSubmit` that rejects with an error that is not a `SubmissionError`, then the error will be written to the console. My opinion is that it is undesirable for the library to do this by default because a rejected promise returned from `onSubmit` is not necessarily something we want our application to consider as an error and write to the console.

**Comment by ClementParis016** [comment]
I totally agree with you @naw, logging the error makes it visible but it doesn't provide any solution to handle it. I would rather like to be able to handle the error myself and report it to an external logging system.

**Comment by naw** [comment]
@ClementParis016 To be clear, you _can_ handle the error yourself now by watching for unhandled promise rejections. This was fixed in #3744 and follows normal promise behavior (except for the extra `console` stuff, of course).  Not sure if this solves your particular problem or not.

**Comment by erikras** [comment]
Published fix in [`v7.3.0`](https://github.com/erikras/redux-form/releases/tag/v7.3.0).
